sequencer/sequencer.go

```diff
@@ -245,6 +249,69 @@ func (s *sequencer) ToggleStep(track, step int) {
 	s.tracks[track].steps[step].clearParameters()
 }
 
+// CopyStep copies a step to the 'clipboard' step.
+// TODO: Double check this, had some AI help with pointers
+func (s *sequencer) CopyStep(track, srcStep int) {
+	if track < 0 || track >= len(s.tracks) || srcStep < 0 || srcStep >= len(s.tracks[track].steps) {
+		return // Out of bounds, do nothing
+	}
+
+	originalStep := s.tracks[track].steps[srcStep]
+
+	// Create a deep copy of the step
+	s.stepClipboard = step{
+		midi:        originalStep.midi, // Assuming midi.Midi is safe to copy directly
```

Comment:
Yep, it's safe. There's only one midi instance and we're passing the pointer around where needed.